### ISSUE TITLE
Unable to install on OS X High Sierra

### ISSUE THREAD

**Issue by dav** [vbmithr/ocaml-ipv6-multicast#3]
It appears three identifiers are not found. I've seem some comments from 2015 that say `MSG_NOSIGNAL` in particular was missing from OS X. Not sure about the others.

```
$ opam install ipv6-multicast
The following actions will be performed:
  ∗  install ipv6-multicast 0.1

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=  🐫 
[ipv6-multicast] Archive in cache

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=  🐫 
[ERROR] The compilation of ipv6-multicast failed at "ocaml pkg/pkg.ml build --pinned false".

#=== ERROR while installing ipv6-multicast.0.1 ================================#
# opam-version 1.2.2
# os           darwin
# command      ocaml pkg/pkg.ml build --pinned false
# path         /Users/dav/.opam/tezos/build/ipv6-multicast.0.1
# compiler     4.04.0
# exit-code    1
# env-file     /Users/dav/.opam/tezos/build/ipv6-multicast.0.1/ipv6-multicast-97288-1872ec.env
# stdout-file  /Users/dav/.opam/tezos/build/ipv6-multicast.0.1/ipv6-multicast-97288-1872ec.out
# stderr-file  /Users/dav/.opam/tezos/build/ipv6-multicast.0.1/ipv6-multicast-97288-1872ec.err
### stdout ###
# src/ipv6_multicast_stubs.c:265:28: error: use of undeclared identifier 'MSG_NOSIGNAL'
# [...]
#                            ^
# src/ipv6_multicast_stubs.c:267:28: error: use of undeclared identifier 'MSG_CMSG_CLOEXEC'
#     case 7: return Val_int(MSG_CMSG_CLOEXEC);
#                            ^
# src/ipv6_multicast_stubs.c:268:28: error: use of undeclared identifier 'MSG_ERRQUEUE'
#     case 8: return Val_int(MSG_ERRQUEUE);
#                            ^
# 7 errors generated.
# Command exited with code 2.
### stderr ###
# pkg.ml: [ERROR] cmd ['ocamlbuild' '-use-ocamlfind' '-classic-display' '-j' '4' '-tag' 'debug'
#      '-build-dir' '_build' '-plugin-tag' 'package(ocb-stubblr)' 'opam'
#      'pkg/META' 'CHANGES.md' 'LICENSE.md' 'README.md' 'src/ipv6_multicast.a'
#      'src/ipv6_multicast.cmxs' 'src/ipv6_multicast.cmxa'
#      'src/ipv6_multicast.cma' 'src/ipv6_multicast.cmx'
#      'src/ipv6_multicast.cmi' 'src/ipv6_multicast.mli'
#      'src/dllipv6_multicast.so' 'src/libipv6_multicast.a']: exited with 10



=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=  🐫 
The following actions failed
  ∗  install ipv6-multicast 0.1
No changes have been performed
```

**Comment by vbmithr** [comment]
I'm don't understand, because when I look at the source code, it seems to me that those undeclared identifies are shielded by a `#ifdef __linux__` which your compiler should not enter if you're under MacOSX. I will need to find a MacOSX machine to try.

Thanks.

**Comment by dav** [comment]
I tried a different laptop that never had OPAM or OCaml installed before, upgraded it to High Sierra and went through the installation again. No problems! It must have been something about upgrading OPAM as well, as I had first installed it over 6 months ago. 

I guess somehow it was trying to compile an older version of the software before the ` #ifdef __linux__` was added?

In any case, sorry to bother. Will close this.
